### PR TITLE
Filers endpoint

### LINKED ISSUE #1507: List of filers - public endpoint
We'll need a public endpoint that returns a list of the filers for a given period.

I think the endpoint should do 2 things:

1. return the full list of filers and
1. allow for a search parameter, eg `?search=<someString>`.

__Would like to hear suggestions for the endpoint URL.__

My only thoughts are a new `/filers` or something off of the existing `/institutions`.

Suggested JSON response:

```json
{
  "institutions": [
    {
      "id": "12345",
      "name": "First Bank",
    },
    {
      "id": "123456",
      "name": "Second Bank",
    }
  ]
}
```

_I don't think any more data is required for now._

### PR DESCRIPTION
**Pull request by jmarin**
Closes #1507

### CODE REVIEW

**Comment by schbetsy** [comment]
@awolfe76 With this PR, the public `/filers` endpoint responses look like this:

```json
{
    "institutions": [
        {
            "institutionId": "6999998",
            "respondentId": "Bank0_RID",
            "period": "2017",
            "name": "bank-0 National Association"
        },
        {
            "institutionId": "testbfl321",
            "respondentId": "testbfl321_RID",
            "period": "2016",
            "name": "Betsy's Filer Test"
        }
    ]
}
```

And it can also filter by filing period (e.g. `/filers/2017`). Does that work for your purposes?

**Comment by schbetsy** [comment]
@jmarin What's the plan for inserting all of the existing filers into `HmdaFilerPersistence`?

File: Documents/admin-api.md [hunk 1]
```diff
@@ -373,3 +373,67 @@ This API is for internal use only. The following endpoints are available
...
+                "period": "2016",
+                "respondentId": "Bank1_RID"
+             }
+        ]
+    }
+    ```
+
+    * `DELETE`
+
+    Deletes HMDA filer.
```
Comment (schbetsy) [inline Documents/admin-api.md]: Please update this to: "Deletes HMDA filer with matching `institutionId`. `name`, `period`, and `respondentId` fields in payload are ignored."

**Comment by awolfe76** [comment]
@schbetsy - Yes, this looks good to me. I've been running it locally to test it out and it's working.

File: persistence-model/src/main/scala/hmda/persistence/messages/commands/institutions/HmdaFilerCommands.scala [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import akka.actor.ActorRef
+import hmda.model.institution.HmdaFiler
+import hmda.persistence.messages.CommonMessages.Command
+
+object HmdaFilerCommands {
+
+  case class CreateHmdaFiler(hmdaFiler: HmdaFiler) extends Command
+  case class CreateHmdaFilerWithReply(hmdaFiler: HmdaFiler, replyTo: ActorRef) extends Command
+  case class DeleteHmdaFiler(hmdaFiler: HmdaFiler) extends Command
+  case class FindHmdaFiler(institutionId: String) extends Command
```
[the remark this answers is not in the document]
Comment (jmarin) [reply persistence-model/src/main/scala/hmda/persistence/messages/commands/institutions/HmdaFilerCommands.scala]: It's being used in the test, see [here](https://github.com/jmarin/hmda-platform/blob/filers-endpoint/persistence/src/test/scala/hmda/persistence/institutions/HmdaFilerPersistenceSpec.scala#L70)

**Comment by schbetsy** [comment]
@jmarin One more question from yesterday: What's the plan for inserting all of the existing filers into `HmdaFilerPersistence`?

**Comment by jmarin** [comment]
@schbetsy The plan for loading data is to use the admin API by creating `HmdaFiler` json objects from the list of institutions that have filed

**Comment by schbetsy** [comment]
Where does that list come from? If we already have a list of institutions that have filed, why do we need `HmdaFilerPersistence` to hold another copy of that list?

**Comment by schbetsy** [comment]
Acutally, never mind. I'll find out when it happens.

**Pull request merged**